src/drivelist.cc

```diff
@@ -0,0 +1,139 @@
+
+  Nan::Set(object, Nan::New<v8::String>("raw").ToLocalChecked(),
+           Nan::New<v8::String>(disk.id).ToLocalChecked());
+
+  Nan::Set(object, Nan::New<v8::String>("description").ToLocalChecked(),
+           Nan::New<v8::String>(disk.caption).ToLocalChecked());
+
+  Nan::Set(object, Nan::New<v8::String>("size").ToLocalChecked(),
+           Nan::New<v8::Number>(disk.size));
+
+  bool isSystem = !disk.removable;
+  bool isProtected = false;
+
+  v8::Local<v8::Object> mountpoints = Nan::New<v8::Array>();
+  for (uint32_t index = 0; index < disk.mountpoints.size(); index++) {
+    if (disk.mountpoints[index].system)
+      isSystem = true;
+    if (disk.mountpoints[index].readonly)
+      isProtected = true;
+    Nan::Set(mountpoints, index, PackMountpoint(disk.mountpoints[index]));
+  }
+
+  Nan::Set(object, Nan::New<v8::String>("system").ToLocalChecked(),
+           Nan::New<v8::Boolean>(isSystem));
+
+  Nan::Set(object, Nan::New<v8::String>("protected").ToLocalChecked(),
+           Nan::New<v8::Boolean>(isProtected));
+
+  Nan::Set(object, Nan::New<v8::String>("mountpoints").ToLocalChecked(),
+           mountpoints);
+
+  return object;
+}
+
+static v8::Local<v8::Object>
+PackDiskList(const std::vector<drivelist::disk_s> &disks) {
+  v8::Local<v8::Object> array = Nan::New<v8::Array>();
+
+  for (uint32_t index = 0; index < disks.size(); index++) {
+    Nan::Set(array, index, PackDisk(disks[index]));
+  }
+
+  return array;
+}
+
+class DrivelistWorker : public Nan::AsyncWorker {
+ public:
+  explicit DrivelistWorker(Nan::Callback *callback)
+    : Nan::AsyncWorker(callback), scanner(), disks() {}
+  ~DrivelistWorker() {}
+
+  void Execute() {
+    drivelist::Code code = this->scanner.Initialize();
+    if (code != drivelist::Code::SUCCESS) {
+      this->SetErrorMessage("Couldn't initialize the scanner");
+      return;
+    }
+
+    code = this->scanner.Scan(&this->disks);
+    if (code != drivelist::Code::SUCCESS) {
+      this->SetErrorMessage("Couldn't scan the drives");
+      return;
+    }
+
+    code = this->scanner.Uninitialize();
+    if (code != drivelist::Code::SUCCESS) {
+      this->SetErrorMessage("Couldn't uninitialize the scanner");
+      return;
+    }
+  }
+
+  void HandleOKCallback() {
+    const unsigned argc = 2;
+    v8::Local<v8::Value> argv[argc] = { Nan::Null(), PackDiskList(disks) };
+    callback->Call(argc, argv);
+  }
+
+ private:
+  drivelist::Scanner scanner;
+  std::vector<drivelist::disk_s> disks;
+};
+
+void List(const Nan::FunctionCallbackInfo<v8::Value> &info) {
+  if (!info[0]->IsFunction()) {
+    return Nan::ThrowTypeError("Callback must be a function");
+  }
+
+  Nan::Callback *callback = new Nan::Callback(info[0].As<v8::Function>());
+  Nan::AsyncQueueWorker(new DrivelistWorker(callback));
+  info.GetReturnValue().SetUndefined();
+}
+
+void DrivelistInit(v8::Local<v8::Object> exports,
+                   v8::Local<v8::Object> module) {
+  Nan::SetMethod(exports, "list", List);
+}
```

Comment:
You should be able to just use the following here instead if defining explicitly:
```cpp
NAN_MODULE_INIT(DrivelistInit) {
  NAN_EXPORT(target, List);
}
```